apps/ejabberd/src/ejabberd_c2s.hrl

```diff
@@ -54,7 +54,7 @@
                 pres_timestamp :: calendar:datetime() | undefined,
                 %% Are we invisible?
                 pres_invis = false :: boolean(),
-                privacy_list = #userlist{} :: mod_privacy:userlist(),
```

Comment:
Actually `mod_privacy:userlist()` is better IMO. Are the definitions different? We tend to reduce use of `.hrl` files whenever it's possible.
Since you introduced module `mognoose_privacy` this type (and other related to privacy) could be move to this module if not `mod_privacy`.